plugin/forward/forward.go

```diff
@@ -65,24 +66,36 @@ func (f *Forward) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg
 	}
 
 	fails := 0
-	var span, child ot.Span
-	var upstreamErr error
-	span = ot.SpanFromContext(ctx)
+	i := 0
+	returnErr := errNoHealthy
+	span := ot.SpanFromContext(ctx)
+	proxyList := f.list()
+	deadline := time.Now().Add(queryTimeout)
+
+	for time.Now().Before(deadline) {
+		if i >= len(proxyList) {
+			if i == fails {
+				// All upstream proxies are dead, assume healtcheck is completely broken and randomly
+				// select an upstream to connect to.
+				i = rand.Intn(len(proxyList))
+				deadline = time.Now() // give it a last attempt
+
+				HealthcheckBrokenCount.Add(1)
```

Comment:
add a `continue` here, so we can drop the else and dedent the body of the else.

Reply:
Well, actually here we detected that all proxies are down, and we need to do the last attempt to connect to random proxy. This logic was in original code. So, I calculate index of random proxy in line 80, specify last attempt by resetting deadline, and go to the main part of loop where proxy is got from list by index and Proxy.connect() is called.
Adding continue at this point will skip connecting to proxy